src/java/main/org/apache/zookeeper/client/StaticHostProvider.java

```diff
@@ -57,26 +62,20 @@
     public StaticHostProvider(Collection<InetSocketAddress> serverAddresses)
             throws UnknownHostException {
         for (InetSocketAddress address : serverAddresses) {
-            InetAddress ia = address.getAddress();
-            InetAddress resolvedAddresses[] = InetAddress.getAllByName((ia!=null) ? ia.getHostAddress():
-                address.getHostName());
+            InetAddress resolvedAddresses[];
+            try {
```

Comment:
@fpj Let's not do this (expose package private JDK methods) given the concerns you raised earlier (as I also commented previously).